### PR TITLE
Feature/spfx yaml pipeline

### PR DESCRIPTION
**Pull request by SebastianSchuetze**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                               |
| New feature?    | no                     |
| New sample/tool?     | yes                               |

## Original Authors
@baywet 

## What's in this Pull Request?
This is an addition to the already existing pipeline for the sampel spfx webpart introduced here https://github.com/SharePoint/sp-dev-build-extensions/pull/18

I added the same pipeline with a yaml-pipeline for Azure DevOps.

@VesaJuvonen this template says, that I should use the dev branch. But the dev branch seems to be outdated and the master branch is much more up-to-date.  So I used that one and also PR to this.

This introduces the improvement that was talked about in #25 

## Guidance

introduces a yaml build pipeline for Azure DevOps for the spfx sample.

### CODE REVIEW

File: samples/azure-devops-ci-cd-spfx/AzureDevOps/build/buildDefinition.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+resources:
+- repo: self
+queue:
+  name: Hosted VS2017
```
Comment (baywet) [inline samples/azure-devops-ci-cd-spfx/AzureDevOps/build/buildDefinition.yaml]: just out of curiosity, have you tried it on other agents (like linux for example) ?

**Comment by baywet** [comment]
Sebastien: you should probably edit your pull request to target dev instead of master, this way we can release all the changes at once with the monthly release.

File: samples/azure-devops-ci-cd-spfx/AzureDevOps/build/buildDefinition.yaml (hunk 1 above)
Comment (SebastianSchuetze) [reply samples/azure-devops-ci-cd-spfx/AzureDevOps/build/buildDefinition.yaml]: Npo I didn't, but I could just run them and try it. Probably that would only work with build tasks, that are programmed with nodejs.

**Comment by SebastianSchuetze** [comment]
will do it. For the target branch. I can only make a new PR. Should I close this one and create a new one and link to the closed one?

**Comment by baywet** [comment]
image
image

**Comment by SebastianSchuetze** [comment]
I did the edit. I think you wanted to achieve Azure DevOps feature to recognize the pipeline automatically, right?

**Comment by baywet** [comment]
thanks! yes, in the case of this specific repository it's not going to work because we're segmented by repository but the hope is that people copy a sample at a time and then it would work.

**Comment by baywet** [comment]
related #25

**Comment by VesaJuvonen** [comment]
thx Vincent

**Pull request merged**